src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -481,9 +481,10 @@ public void setHomeBuilding(@Nullable final AbstractBuilding building)
         if (homeBuilding != null && building != null && !homeBuilding.equals(building))
         {
             homeBuilding.removeCitizen(this);
+            markDirty();
         }
 
-        if (building instanceof BuildingHome || building instanceof BuildingBarracksTower)
+        if (building == null || building instanceof BuildingHome || building instanceof BuildingBarracksTower)
```

Comment:
building == null is irrelevant here and an additional check that is not needed. Null is never an instance of.

Reply:
Its not, three types of buildings can be assigned to citizens:
BuildingHome, BuildingBarracksTower and null (removing them from a building)

Reply:
Ah sorry misread it.